### ISSUE TITLE
[$500] mWeb/Chrome - Profile setting - The autofill field remains when scrolling the page

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11336]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with any account
3. Go to Settings->Profile
4. Focus on the name input field (first or last name)
5. Scroll the page

## Expected Result:
The autofill field should be hidden

## Actual Result:
The autofill field remains

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Mobile Web

**Version Number:** 1.2.7.2

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/192529923-4a82354a-56d4-4971-869f-162b25da8ab0.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by marcochavezf** [comment]
I was able to reproduce this one on production too. Looking at other sites, the autofill pop-up disappears when the user scrolls down/up the page.

**Comment by tienifr** [comment]
# Proposal

## Problem
The default behavior of removing autofill is when we scroll `window`. Unfortunately, our form stays inside other container so when we scroll, this container scroll not `window`.

ref: https://stackoverflow.com/a/45446880

## Solution
My suggestion is to remove and restore from the active input field in a form's onScroll event

in https://github.com/Expensify/App/blob/1bed3e5bc8c895eb630206e19c0aa82970f890f1/src/pages/settings/Profile/ProfilePage.js

```diff
+ 
+    onScroll(){
+       const a = document.activeElement;
+        a.blur();
+       a.focus();
+   }
```

```diff
                <Form
                    style={[styles.flexGrow1, styles.ph5]}
                    formID={CONST.PROFILE_SETTINGS_FORM}
                    validate={this.validate}
                    onSubmit={this.updatePersonalDetails}
                    submitButtonText={this.props.translate('common.save')}
+                  onScroll={this.onScroll}
                >
```

In https://github.com/Expensify/App/blob/1bed3e5bc8c895eb630206e19c0aa82970f890f1/src/components/Form.js

```diff
    onSubmit: PropTypes.func.isRequired,

    children: PropTypes.node.isRequired,

+    onScroll: PropTypes.func,

```

```diff
                <ScrollView
                    style={[styles.w100, styles.flex1]}
                    contentContainerStyle={styles.flexGrow1}
                    keyboardShouldPersistTaps="handled"
+                  onScroll={this.props.onScroll}
                >
```


https://user-images.githubusercontent.com/113963320/192963260-67aba93e-5c88-4ce8-9a0a-05104d51eda5.mp4

**Comment by mananjadhav** [comment]
Waiting for better proposals.

**Comment by marcochavezf** [comment]
> @tienifr Pardon my ignorance on the topic, but I feel there must be a better way to handle this. We're using RN components and I don't think we should be modify the scroll event, especially with a window object.
> 
> What do you think? @marcochavezf

Yeah, using `window` and `scrollEventThrottle` looks a bit hacky for me. I think the problem could be related to one HTML property or something, I don't think other sites solve this issue by modifying the scroll event.

**Comment by dylanexpensify** [comment]
@mananjadhav should we double

**Comment by mananjadhav** [comment]
> @mananjadhav should we double

Yeah @dylanexpensify, we should.

**Comment by dylanexpensify** [comment]
updated/doubled:
https://www.upwork.com/ab/applicants/1578314536620560384/job-details
https://www.upwork.com/jobs/~015fdd2869753011b3

**Comment by AmroElzeiny** [comment]
I think we may edit the scroll event property, that is when the mouse leaves the input or on-blur or on-scrolling we will hide the autofill menu! I think it's as simple as it is, with no complexity

**Comment by getusha** [comment]
I don't think the auto fill has to be hidden, because the input field is still focused.
Instead what we have to do is just make the auto fill stay with the input field when it is scrolled.
Does this sound right?

**Comment by mananjadhav** [comment]
Thanks for the comment @AmroElzeiny. Are you trying to post a proposal for the job? Can you elaborate on the technical side of the implementation? Which files/code will be updated? With what changes?

@getusha Making autofill stay should also be acceptable in my opinion. What do you think @dylanexpensify @marcochavezf

**Comment by AmroElzeiny** [comment]
I'm sorry but I don't know who I'm talking to regarding this project. @getusha or @mananjadhav ?

**Comment by mananjadhav** [comment]
@AmroElzeiny You're talking to me at the moment or everyone assigned to the issue on the top right.

But we always welcome contributors to propose solutions on the job, share feedback, etc.

**Comment by AmroElzeiny** [comment]
I have only one question, Mr Manan, you want the autofill menu to remain in its position stuck to the input, right?

**Comment by getusha** [comment]
@marcochavezf @mananjadhav 
 go to https://staging.new.expensify.com/ on the login page
-  zoom in until you get the scroll bar
-  focus the email text field 
- scroll and you see the tooltip hides

The autofill hides itself when it detects a scroll on the body ( not custom overflows on div )
I have a fix for this but the outcome is it completely removes focus from the input is that okay?

**Comment by getusha** [comment]
# Proposal
> **This is _the best_ and the _only solution in RN Applications** 

..........

> src/pages/settings/Profile/ProfilePage.js
```javascript
//save the targeted input field to a state
setFocusedInput(focused){
    this.setState({focusedInput: focused.target})
}
```
```javascript
<Form
// focusedInput prop passes the targeted input field
   focusedInput={this.state.focusedInput} >
  
<TextInput
// set focusedInput state depending on the focused input field
   onFocus={(focused)=>{this.setFocusedInput(focused)}}
```

> src/components/Form.js
```javascript
onScroll={()=>{
   this.props.focusedInput.blur()
   this.props.focusedInput.focus()
}}

```
## **Final Result**


https://user-images.githubusercontent.com/75031127/194609325-8c4dfa17-1ed2-4439-817b-bd9941ffbad5.mp4

>_using window is not acceptable way of approaching this problem, the above implementation fixes the issue perfectly with the right way_

**Comment by dylanexpensify** [comment]
@marcochavezf thoughts?

**Comment by mananjadhav** [comment]
I went through @aimane-chnaif's proposal and I am almost inclined to agree with it. I haven't tested it. But I would want someone from the @Expensify/design  team to see if there's any issue with changing the cardStyle here.

**Comment by shawnborton** [comment]
The proposal seems good and thorough but given that it has larger implications, we should really test it out on all flows. 

Where else is cardStyle used? Also, that name is so generic it's hard to understand what cardStyle even refers to.

**Comment by marcochavezf** [comment]
> I went through @aimane-chnaif's proposal and I am almost inclined to agree with it. I haven't tested it. But I would want someone from the @Expensify/design team to see if there's any issue with changing the cardStyle here.

Yeah, the proposal also looks promising to me, and I agree that the solution should be tested on all `AuthScreens` (I wonder why `height: 100%` was set manually before).

**Comment by dylanexpensify** [comment]
@aimane-chnaif any update?

**Comment by marcochavezf** [comment]
Ah just to follow the established process, @mananjadhav do you agree with the solution? Can we consider it as C+ reviewed?

> I went through @aimane-chnaif's proposal and I am almost inclined to agree with it. I haven't tested it. But I would want someone from the @Expensify/design team to see if there's any issue with changing the cardStyle here.

**Comment by mananjadhav** [comment]
Yeah we can consider this C+ Reviewed for @aimane-chnaif's proposal

**Comment by marcochavezf** [comment]
Cool, thanks for the clarification @mananjadhav! Assigning @aimane-chnaif 🚀

**Comment by mananjadhav** [comment]
Thanks for the update @aimane-chnaif

**Comment by mananjadhav** [comment]
Waiting for an update here @aimane-chnaif

**Comment by mananjadhav** [comment]
@aimane-chnaif Been two weeks waiting on this one. Please provide an update.

**Comment by mananjadhav** [comment]
@aimane-chnaif Please send us an update within 24 hours, else we'll have to rescind the job here.

**Comment by marcochavezf** [comment]
> Sorry for late but it took much efforts to test and fix all regressions for all auth screens on all platforms. Especially all modal pages with transition animations and with input autoFocus enabled. I raised [PR](https://github.com/Expensify/App/pull/12366). Since this is navigation card and animation styles refactor, all screens need to be throughly tested throughout the app in all devices and all platforms.

Thanks for the update @aimane-chnaif and also thanks for testing it thoughtfully! Next time just a minor update like "still testing", "found x issue" or a list of what you have been testing work for us. We understand there are a lot of places that could be impacted by this change :)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by dylanexpensify** [comment]
Can we get an update here @mananjadhav @aimane-chnaif

**Comment by mananjadhav** [comment]
Reviewed the PR. Raised some questions and asked for videos instead of images. Will test once @aimane-chnaif responds.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mananjadhav** [comment]
@dylanexpensify This is pending with me, I am currently testing it.

**Comment by dylanexpensify** [comment]
Any ETA on when you'll be done @mananjadhav ?

**Comment by mananjadhav** [comment]
Targeting to finish this by EOD today

**Comment by mananjadhav** [comment]
Apologies for the time taken to review. Required testing across multiple screens. This is C+ reviewed now and approved.

**Comment by aldo-expensify** [comment]
Reported by @Puneet-here , the PR https://github.com/Expensify/App/pull/12366 caused the regressions:

- https://github.com/Expensify/App/issues/13011

- https://github.com/Expensify/App/issues/13008

**Comment by aldo-expensify** [comment]
> @aldo-expensify the Next button was missing on split bill page.
> 
> This shouldn't have passed QA, right? Maybe we should write automated tests for this since it made to production.

I think it is a good call, we should review our manual tests because this could have been catch. I don't think it would be easy to do as an automated test because the button is not completely missing, it is just at the bottom of the list instead of at the bottom of the screen (fixed).

<img width="452" alt="image" src="https://user-images.githubusercontent.com/87341702/204043801-6421c8ad-8680-4654-9c40-8c9edc5d925b.png">

@isagoico can we make sure that in the manual test for the split bill we test:
- on mobile web, using a "high traffic account" (with enough contacts to have scrolling), make sure that the "Next" button appears **fixed at the bottom of the screen** after selecting the first participant?

**Comment by mananjadhav** [comment]
@aimane-chnaif Can we look into the reported regressions and fix up with a new PR?

**Comment by mananjadhav** [comment]
> @aimane-chnaif Can we look into the reported regressions and fix up with a new PR?

@aimane-chnaif Did you get a chance to look at the comment?

**Comment by mananjadhav** [comment]
Okay. Thanks for the update.

**Comment by dylanexpensify** [comment]
Working on this now to discuss Regression testing steps

**Comment by dylanexpensify** [comment]
Posted in internal channel on how to align on regression steps (cc @JmillsExpensify). 

@aimane-chnaif when can we expect an updated PR?

**Comment by hellohublot** [comment]
# Proposal

### How about this

https://github.com/Expensify/App/blob/bcdb1413fab510ee97c38928aba053b37b1d46d9/src/components/Form.js#L248-L252
```diff
<ScrollView
     ...
+    keyboardDismissMode={this.props.keyboardDismissMode}
/>
```

https://github.com/Expensify/App/blob/bcdb1413fab510ee97c38928aba053b37b1d46d9/src/pages/settings/Profile/ProfilePage.js#L231-L238

```diff
<Form
     ...
+    keyboardDismissMode={'on-drag'}
/>
```

**Comment by dylanexpensify** [comment]
@marcochavezf @mananjadhav thoughts on @aimane-chnaif's comment?

**Comment by mananjadhav** [comment]
@hellohublot Does the `keyboardDismissMode` blur the input or something that autofill will hide? Based on what I saw in `react-native-web`, it just dismisses the keyboard. I am not sure if we want that. The user might scroll and want to focus on the next input, where the keyboard toggles and that's a bad UX.

@dylanexpensify @marcochavezf I did spend sometime (not more than 2 hours) looking at the regressions, and the solution we worked on earlier. I couldn't come up with a solid approach that fixes all of the issues. Looking at the ROI of the bug, I think we should can ignore and _do nothing_ it.

**Comment by marcochavezf** [comment]
Yeah it's a weird issue, because in the SignIn page is working fine (RootStack/Public screen):

https://user-images.githubusercontent.com/6829422/205416127-9f1382b3-8896-4bee-afed-7930177af2d2.mov

I think we'd need to come up with a holistic solution to standardize the card styles for both ModalStackNavigator and RootStack screens, but I agree that solving this issue is more a "nice to have" (I initially thought the solution was going to be simple). IMO we can close this one atm to focus on other significant issues and re-open it if there are more issues related.

**Comment by dylanexpensify** [comment]
Sounds good, let's close!
